src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs

```diff
@@ -218,19 +222,48 @@ private void UpdateCanvasSize(ShadowCollection? shadows)
 		double newHostHeight = _currentContent.ActualHeight + newHostSpreedHeight * 2;
 		double newHostWidth = _currentContent.ActualWidth + newHostSpreedWidth * 2;
 
-		var left = -(newHostSpreedWidth
-										+ (_currentContent.HorizontalAlignment == HorizontalAlignment.Left ? -_currentContent.Margin.Left : 0)
-										+ (_currentContent.HorizontalAlignment == HorizontalAlignment.Right && _currentContent.VerticalAlignment != VerticalAlignment.Center ? _currentContent.Margin.Right == 0 ? 0 :
-														_currentContent.ActualWidth + _currentContent.Margin.Right - _canvas.Margin.Right : 0)
-										+ (_currentContent.HorizontalAlignment == HorizontalAlignment.Stretch ? _currentContent.Margin.Right : 0)
-										+ (_currentContent.HorizontalAlignment == HorizontalAlignment.Center ? _currentContent.Margin.Left : 0)
-										);
-		var top = -(newHostSpreedHeight
-										+ (_currentContent.VerticalAlignment == VerticalAlignment.Top ? -_currentContent.Margin.Top : 0)
-										+ (_currentContent.VerticalAlignment == VerticalAlignment.Bottom ? +_currentContent.Margin.Bottom + _currentContent.ActualHeight : 0)
-										+ (_currentContent.VerticalAlignment == VerticalAlignment.Stretch ? _currentContent.Margin.Bottom : 0)
-										+ (_currentContent.VerticalAlignment == VerticalAlignment.Center ? _currentContent.Margin.Top : 0)
+		double top = 0;
+		double left = 0;
+
+		if (_currentContent.VerticalAlignment == VerticalAlignment.Center)
+		{
+			_canvasInner.Margin = _canvas.Margin = _currentContent.Margin;
+			_canvasInner.Margin = _canvas.Margin = _currentContent.Margin;
+
+			if (_currentContent.Margin == new Thickness(0, 0, 0, 0))
```

Comment:
Here I think you can use the `Empty` value from Thickness
```suggestion
			if (_currentContent.Margin == Thickness.Empty)
```